android/app/src/main/java/org/mozilla/magnet/db/History.java

```diff
@@ -0,0 +1,192 @@
+package org.mozilla.magnet.db;
+
+import android.content.ContentValues;
+import android.content.Context;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.database.sqlite.SQLiteOpenHelper;
+import android.util.Log;
+
+import org.mozilla.magnet.db.Schema.HistoryEntry;
+
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
+
+import static android.provider.BaseColumns._ID;
+import static org.mozilla.magnet.db.Schema.HistoryEntry.COLUMN_NAME_TIME_LAST_SEEN;
+import static org.mozilla.magnet.db.Schema.HistoryEntry.COLUMN_NAME_URL;
+import static org.mozilla.magnet.db.Schema.HistoryEntry.COLUMN_NAME_TIME_FIRST_SEEN;
+
+public class History extends SQLiteOpenHelper {
```

Comment:
Can we change the approach and make History not coupled with the SQLiteOpenHelper?

My point is the History class will have some methods like record, update, updateLastSeen, that has some semantic meanining. That should not be tied to the use of SQLite, we could pass a dependency in the constructor of History, that does the actual data saving and processing. That will help us to change the implementation for any other if needed and better unit tests when we have them :)

For example:
```java
public class History {
    public History(Context context, DataModel model) {}
}

public interface DataModel {
    public void onCreate() {}
    public void onUpgrade() {}
    public void insert() {}
....
}

public class SQLiteDataModel implements DataModel {
...
}
,,,

Also we could have a factory to get the History objects already initialized with the strategy we want:

```java
public class HistoryFactory {
     public static History getHistory(Context ctx) {
        return new History(ctx, new SQLiteDataModel());
        // Or even better a singleton.
     }
}